### PR TITLE
Keep SM2 wrap in EC to be compatible with common implementations

### PR DESCRIPTION
**Pull request by dongbeiouba**
Fix the incompatibility issue for SM2.
When the Public Key Algorithm which in SubjectPublicKeyInfo for SM2 cert is sm2,
it's not recognized by most implementations, should modify the Public Key
Algorithm from sm2 to id-ecPublicKey for signing new SM2 certs.

Actually, SM2 can be wrapped by EC with the EC OID and EC pem header. So if we
see EC OID and literal, we try both key types.

Also support SM2 private key with "BEGIN SM2 PRIVATE KEY".

##### Checklist

- [ ] 在 https://yuque.com/tsdoc 增加或更新了必要的文档
- [x] 增加或更新了必要的测试用例
- [ ] 对于重要修改，更新了CHANGES文件
- [ ] 当前修改存在对已有API参数或返回值的改变
- [ ] 当前修改存在对旧版本功能的兼容性改变（如网络协议或密码算法）

### CODE REVIEW

**Review by wa5i — DISMISSED** [review]
LGTM

**Comment by wa5i** [comment]
CI 跪了

File: providers/implementations/encode_decode/encode_key2any.c [hunk 1]
```diff
@@ -808,10 +804,6 @@ static int ecx_pki_priv_to_der(const void *vecxkey, unsigned char **pder)
 # define ed448_evp_type         EVP_PKEY_ED448
 # define x25519_evp_type        EVP_PKEY_X25519
 # define x448_evp_type          EVP_PKEY_X448
-# define ed25519_input_type     "ED25519"
-# define ed448_input_type       "ED448"
-# define x25519_input_type      "X25519"
-# define x448_input_type        "X448"
```
Comment (wa5i) [inline providers/implementations/encode_decode/encode_key2any.c:814]: 为啥删除这几个宏？

Comment (dongbeiouba) [reply providers/implementations/encode_decode/encode_key2any.c:814]: > 为啥删除这几个宏？

dead code

**Review by wa5i — APPROVED** [review]

**Review by InfoHunter — APPROVED** [review]

**Pull request merged**